In [1]:
import pandas as pd

print("Pandas imported successfully!")

Pandas imported successfully!


In [2]:
student_data = {
    "Name": ["Srujan", "Rahul", "Ananya", "Priya", "Kiran"],
    "Age": [22, 21, 22, 20, 23],
    "Marks": [85, 78, 92, 88, 75],
    "Department": ["ISE", "CSE", "ISE", "ECE", "CSE"]
}

student_df = pd.DataFrame(student_data)

student_df

,Name,Age,Marks,Department
0,Srujan,22,85,ISE
1,Rahul,21,78,CSE
2,Ananya,22,92,ISE
3,Priya,20,88,ECE
4,Kiran,23,75,CSE


In [3]:
from sklearn.datasets import load_iris

iris = load_iris()

iris_df = pd.DataFrame(
    iris.data,
    columns=iris.feature_names
)

iris_df["species"] = [
    iris.target_names[i] for i in iris.target
]

iris_df.head()

,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


In [4]:
print("First 5 rows:")
iris_df.head()

First 5 rows:


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


In [5]:
print("Shape of dataset:", iris_df.shape)

Shape of dataset: (150, 5)


In [6]:
print("Columns:")
print(iris_df.columns.tolist())

Columns:
['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)', 'species']


In [7]:
print("Data Types:")
print(iris_df.dtypes)

Data Types:
sepal length (cm)    float64
sepal width (cm)     float64
petal length (cm)    float64
petal width (cm)     float64
species               object
dtype: object


In [8]:
print("Dataset Information:")
iris_df.info()

Dataset Information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   species            150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [9]:
print("Summary Statistics:")
iris_df.describe()

Summary Statistics:


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm)
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333
std,0.828066,0.435866,1.765298,0.762238
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


In [10]:
print("Number of rows:", iris_df.shape[0])
print("Number of columns:", iris_df.shape[1])
print("Total elements:", iris_df.size)

Number of rows: 150
Number of columns: 5
Total elements: 750


## Observations

The Iris dataset contains 150 rows and 5 columns.

The dataset contains four numerical features and one categorical species column.

The `head()` function was used to view the first few records.

The `info()` function was used to inspect the structure, non-null values, and data types.

The `describe()` function was used to obtain summary statistics for numerical variables.

The DataFrame was also explored using `shape`, `columns`, `dtypes`, and `size`.

## Interview Questions

### 1. What is a Pandas DataFrame?

A Pandas DataFrame is a two-dimensional labeled data structure in Python. It stores data in rows and columns, similar to a table or spreadsheet.

### 2. What information does `info()` provide?

The `info()` function provides information about the DataFrame structure, including the number of rows, column names, non-null values, data types, and memory usage.

### 3. What is the difference between `shape` and `size`?

`shape` returns the number of rows and columns in a DataFrame.

For example:

`(150, 5)` means 150 rows and 5 columns.

`size` returns the total number of elements in the DataFrame.

For a DataFrame with 150 rows and 5 columns:

`150 × 5 = 750` elements.